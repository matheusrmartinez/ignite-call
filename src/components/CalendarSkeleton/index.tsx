import { CaretLeft, CaretRight } from 'phosphor-react'
import {
  CalendarContainer,
  CalendarHeader,
  CalendarTitle,
  CalendarActions,
  CalendarBody,
  CalendarDay,
} from './styles'

export const CalendarSkeleton = () => {
  const weekDays = ['DOM', 'SEG', 'TER', 'QUA', 'QUI', 'SEX', 'SAB']

  const calendarWeeks = [
    { week: 0, days: [1, 2, 3, 4, 5, 6, 7] },
    { week: 1, days: [1, 2, 3, 4, 5, 6, 7] },
    { week: 2, days: [1, 2, 3, 4, 5, 6, 7] },
    { week: 3, days: [1, 2, 3, 4, 5, 6, 7] },
    { week: 4, days: [1, 2, 3, 4, 5, 6, 7] },
  ]

  return (
    <CalendarContainer>
      <CalendarHeader>
        <CalendarTitle>
          <span></span>
        </CalendarTitle>
        <CalendarActions>
          <button>
            <CaretLeft />
          </button>
          <button>
            <CaretRight />
          </button>
        </CalendarActions>
      </CalendarHeader>

      <CalendarBody>
        <thead>
          <tr>
            {weekDays.map((weekDay) => {
              return <th key={weekDay}>{weekDay}.</th>
            })}
          </tr>
        </thead>
        <tbody>
          {calendarWeeks.map(({ week, days }) => (
            <tr key={week}>
              {days.map((_, index) => (
                <td key={index}>
                  <CalendarDay />
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </CalendarBody>
    </CalendarContainer>
  )
}
