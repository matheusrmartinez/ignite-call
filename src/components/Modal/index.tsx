import { ModalContainer } from './styles'

interface ModalProps {
  children: JSX.Element
}

export const Modal = ({ children }: ModalProps): JSX.Element => {
  return <ModalContainer>{children}</ModalContainer>
}
