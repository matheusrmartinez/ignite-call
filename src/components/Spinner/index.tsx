import { SpinnerContainer } from './styles'

export const Spinner = () => {
  return <SpinnerContainer />
}
