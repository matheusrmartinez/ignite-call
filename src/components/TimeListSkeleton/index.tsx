import { TimePickerItem, TimePickerList } from './styles'

export const TimeListSkeleton = () => {
  const timeList: string[] = Array.from({ length: 9 })

  return (
    <TimePickerList>
      {timeList.map((_, index) => {
        return <TimePickerItem key={index} />
      })}
    </TimePickerList>
  )
}
