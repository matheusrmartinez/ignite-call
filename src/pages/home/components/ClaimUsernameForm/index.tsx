import { Button, TextInput, Text } from "@ignite-ui/react";
import { Form, FormAnnotation } from "./styles";
import { ArrowRight } from "phosphor-react";
import { useForm } from "react-hook-form";
import { z } from "zod";
import { zodResolver } from "@hookform/resolvers/zod";

const claimUsernameFormSchema = z.object({
  username: z
    .string()
    .min(3, { message: "O usuário precisa ter pelo menos 3 letras" })
    .regex(/^([a-z\\\\-]+)$/i, {
      message: "O usuário pode ter apenas números e hífens",
    })
    .transform((username) => username.toLowerCase()),
});

type ClaimUsernameFormData = z.infer<typeof claimUsernameFormSchema>;

export const ClaimUsernameForm = () => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<ClaimUsernameFormData>({
    resolver: zodResolver(claimUsernameFormSchema),
  });

  const handleClaimUsername = (data: ClaimUsernameFormData) => {
    console.log(data.username);
  };

  return (
    <>
      <Form as="form" onSubmit={handleSubmit(handleClaimUsername)}>
        <TextInput
          size="sm"
          prefix="ignite.com/"
          placeholder="seu-usuário"
          {...register("username")}
        />
        <Button size="sm" type="submit">
          Reservar
          <ArrowRight />
        </Button>
      </Form>
      <FormAnnotation>
        <Text size="sm">
          {errors.username
            ? errors.username.message
            : "Digite o nome de usuário desejado"}
        </Text>
      </FormAnnotation>
    </>
  );
};
