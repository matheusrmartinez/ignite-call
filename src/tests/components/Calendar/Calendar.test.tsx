import { Calendar } from '@/components/Calendar'
import { mockedBlockedDates } from '@/tests/__mocks__/mockedBlockedDates'
import { render } from '@testing-library/react'

function getBlockedDates() {
  return mockedBlockedDates
}

jest.mock('next/router', () => ({
  useRouter: () => ({
    asPath: '/',
    replace: jest.fn(),
    query: { username: 'matheusmartinez' },
  }),
}))

jest.mock('@tanstack/react-query', () => ({
  useQuery: jest.fn().mockReturnValue({
    data: { blockedWeekDays: [0, 3, 4, 5, 6], blockedDates: [9] },
    isLoading: false,
    error: {},
  }),
}))

describe('Home', () => {
  const onDateSelected = jest.fn()

  it('renders homepage unchanged', () => {
    const { container } = render(<Calendar onDateSelected={onDateSelected} />)
    expect(container).toMatchSnapshot()
  })
})
