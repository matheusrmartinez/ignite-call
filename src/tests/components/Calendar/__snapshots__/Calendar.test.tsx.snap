// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div
    class="c-huPADD"
    data-testid="calendar-container"
  >
    <div
      class="c-jVwFOG"
    >
      <p
        class="c-kQRPbk c-fyeWXm c-kQRPbk-kWyXas-size-md"
      >
        May
         
        <span>
          2023
        </span>
      </p>
      <div
        class="c-lbXbSD"
      >
        <button
          title="Previous month"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <polyline
              fill="none"
              points="160 208 80 128 160 48"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
        </button>
        <button
          title="Next month"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <polyline
              fill="none"
              points="96 48 176 128 96 208"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
        </button>
      </div>
    </div>
    <table
      class="c-cpSFgM"
    >
      <thead>
        <tr>
          <th>
            DOM
            .
          </th>
          <th>
            SEG
            .
          </th>
          <th>
            TER
            .
          </th>
          <th>
            QUA
            .
          </th>
          <th>
            QUI
            .
          </th>
          <th>
            SEX
            .
          </th>
          <th>
            S√ÅB
            .
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              30
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              1
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              2
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              3
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              4
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              5
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              6
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              7
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              8
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              9
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              10
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              11
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              12
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              13
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              14
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              15
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              16
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              17
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              18
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              19
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              20
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              21
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
            >
              22
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
            >
              23
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              24
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              25
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              26
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              27
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              28
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
            >
              29
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
            >
              30
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              31
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              1
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              2
            </button>
          </td>
          <td>
            <button
              class="c-bgiKMN"
              disabled=""
            >
              3
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
